include/exadg/incompressible_navier_stokes/spatial_discretization/calculators/wall_shear_stress_calculator.cpp

```diff
@@ -0,0 +1,233 @@
+
+        for(auto const i : fe_face_values.quadrature_point_indices())
+        {
+          dealii::Point<dim> const point_face = fe_face_values.quadrature_point(i);
+          for(auto const j : fe_values.quadrature_point_indices())
+          {
+            if(point_face.distance_square(fe_values.quadrature_point(j)) < tol_sqrd)
+            {
+              face_cell_index_matches[i] = j;
+              break;
+            }
+          }
+        }
+
+        face_to_cell_index[face] = face_cell_index_matches;
+      }
+
+      break;
+    }
+  }
+}
+
+template<int dim, typename Number>
+void
+WallShearStressCalculator<dim, Number>::compute_wall_shear_stress(
+  VectorType &                                dst,
+  VectorType const &                          src,
+  std::shared_ptr<dealii::Mapping<dim> const> mapping,
+  std::set<dealii::types::boundary_id> const  write_wall_shear_stress_boundary_IDs) const
+{
+  dst = 0;
+
+  // if no write_wall_shear_stress_boundary_IDs is provided, the default invalid_boundary_id
+  // indicates writing the wall_shear_stress on all boundaries.
+  bool write_on_all_boundary_IDs = false;
+  if(write_wall_shear_stress_boundary_IDs.size() == 1)
+  {
+    // no boundary_id(s) provided, but write_wall_shear_stress == true
+    if(*write_wall_shear_stress_boundary_IDs.begin() == dealii::numbers::invalid_boundary_id)
+      write_on_all_boundary_IDs = true;
+  }
+
+  bool is_hyper_cube =
+    matrix_free->get_dof_handler(dof_index).get_fe(0).reference_cell().is_hyper_cube();
+  AssertThrow(is_hyper_cube,
+              dealii::ExcMessage("WallShearStressCalculator assumes hypercube reference cells."));
+
+  unsigned int const    fe_degree = matrix_free->get_dof_handler(dof_index).get_fe(0).degree;
+  dealii::FESystem<dim> fe_system(dealii::FE_DGQ<dim>(fe_degree), dim);
+  dealii::QGaussLobatto<dim - 1> face_quadrature(fe_degree + 1);
+  dealii::QGaussLobatto<dim>     quadrature(fe_degree + 1);
+
+  unsigned int const dofs_per_component =
+    matrix_free->get_dof_handler(dof_index).get_fe(0).dofs_per_cell / dim;
+
+  unsigned int const dofs_per_comp =
+    matrix_free->get_dof_handler(dof_index).get_fe(0).base_element(0).dofs_per_cell;
+
+#ifdef DEBUG
+  dealii::FEFaceValues<dim> fe_face_values(*mapping,
+                                           fe_system,
+                                           face_quadrature,
+                                           dealii::update_quadrature_points |
+                                             dealii::update_gradients |
+                                             dealii::update_normal_vectors);
+  dealii::FEValues<dim>     fe_values(*mapping,
+                                  fe_system,
+                                  quadrature,
+                                  dealii::update_quadrature_points);
+#else
+  dealii::FEFaceValues<dim> fe_face_values(
+    *mapping, fe_system, face_quadrature, dealii::update_gradients | dealii::update_normal_vectors);
```

Comment:
This was not leading to an error, since I only used 'FE_DGQ', I did not test with other element types, which is of course very restrictive - I will address this in the other comment section.